### PR TITLE
Add river adjacency matrix

### LINKED ISSUE #169: Update adjacency matrix
The matrix we pulled from a ScienceBase entry has 3 reaches missing. This issue is to generate a new adjacency matrix that contains those 3 reaches.

### PR DESCRIPTION
**Pull request by lekoenig**
This PR adds the subseg adjacency matrix from the DRB temperature quest to our project pipeline. `1_fetch/in/subseg_distance_matrix.rds` was downloaded from the shared temperature data cache and is committed here. The resulting object is a list containing four different versions of the distance matrix. I've created a separate target called `p1_drb_distance_matrix_updown` in case we opt to use other versions of the distance matrix down the line.

I haven't yet run the pipeline to push these targets to S3. @jds485, would you mind updating the targets meta file? Or should I go ahead and commit/push my version?

Closes #169

### CODE REVIEW

**Comment by jds485** [comment]
Looks good! I updated the targets and also made a csv file of the distance matrix that can be read into Python. I think this can be merged after you review what I added

File: 1_fetch.R
Comment (lekoenig) [inline 1_fetch.R:461]: Oh, great addition - I wasn't thinking about the need to export for python!

One small comment that writing the file path twice increases the likelihood of making a typo - this looks good as you've written it but here's a proposed edit I think we've used elsewhere in this pipeline:

```suggestion
    {
      fileout <- "1_fetch/out/drb_distance_matrix_updown.csv"
      write.csv(x = as.data.frame(p1_drb_distance_matrix$updown), 
              file = fileout,
              row.names = TRUE)
      fileout
    },
```

**Comment by jds485** [comment]
Thanks! Before merging, I want to double check the format expected by river-dl (I think it might be a numpy array instead of csv, and I want to also check the format of the rows and columns). I may add one more commit with a target that writes a numpy array.

**Comment by jds485** [comment]
I added the npz file generation target

File: 2_process/src/write_data.R
Comment (lekoenig) [inline 2_process/src/write_data.R:41]: What do you think of adding a brief comment here that we're expecting a list containing four matrices named "complete," "downstream," "upstream," and "updown?"

File: 1_fetch.R
Comment (lekoenig) [inline 1_fetch.R:454]: I was wondering whether we still want to keep this csv file target, but I guess it doesn't hurt to keep as a way of quickly checking the format/dimensions of the updown matrix.

File: 1_fetch.R
Comment (lekoenig) [inline 1_fetch.R:467]: Thanks for adding the npz target! Do any of our other inputs need to be formatted as npz, or just the distance matrix?

Comment (jds485) [reply 1_fetch.R:467]: Just the distance matrix. The other inputs are attributes and SC observations, which are both saved as zarr files.

File: 2_process/src/write_data.R
Comment (jds485) [reply 2_process/src/write_data.R:41]: Thanks! I meant to add that comment. I could not figure out how to get the `keys` automatically recognized within `savez_compressed`. In Python, a named dict is automatically converted to have the same names in the npz file, but that did not happen using reticulate

**Comment by jds485** [comment]
@lekoenig I think I addressed your comments. Let me know if this is good to merge. I approved based on your edits

**Review by jds485 — APPROVED** [review]

File: 1_fetch.R
Comment (lekoenig) [reply 1_fetch.R:467]: Sounds good, thanks for the reminder about the expected file formats!

**Comment by lekoenig** [comment]
Looks great, @jds485! I'll defer to you to merge whenever you think it's ready. I tried to "approve" the PR, but GitHub apparently won't let me do that on my own PR 😄

**Pull request merged**
